### PR TITLE
Database Support

### PR DESCRIPTION
**Pull request by GeetikaBatra**

### CODE REVIEW

**Comment by msrb** [comment]
@GeetikaBatra please let me know when this PR is ready to be reviewed :)

**Comment by GeetikaBatra** [comment]
Sure @msrb  :D

File: src/errors.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Author: Geetika Batra <gbatra@redhat.com>
+#
+
+"""Declaration of classes representing various exception types."""
+
+
+class TaskError(Exception):
```
Comment (msrb) [inline src/errors.py]: Do we need this?

File: src/models.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+class Version(Base):
+    __tablename__ = 'versions'
+    # package_id together with version identifier must be unique
+    __table_args__ = (UniqueConstraint(
+        'package_id', 'identifier', name='pv_unique'),)
+
+    id = Column(Integer, primary_key=True)
+    package_id = Column(Integer, ForeignKey(Package.id))
+    identifier = Column(String(255), index=True)
+    synced2graph = Column(Boolean, nullable=False, default=False)
```
Comment (msrb) [inline src/models.py]: Do we need this?

File: src/models.py [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+    package = relationship(Package, back_populates='versions', lazy='joined')
+    analyses = relationship('Analysis', back_populates='version')
```
Comment (msrb) [inline src/models.py]: There is no `Analysis` table

**Comment by GeetikaBatra** [comment]
Thanks @msrb, fixing them :D

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+ENV LANG=en_US.UTF-8 \
+    # place for alembic migrations
+    ALEMBIC_DIR='/alembic'
+RUN yum install -y epel-release &&\
+    yum install -y gcc  python34-pip python34-requests python34-devel &&\
+    yum clean all
+
+RUN pip3 install --upgrade pip>=10.0.0
+
+# Make sure random user has place to store files
```
[the remark this answers is not in the document]
Comment (msrb) [reply Dockerfile]: Please remove this line the your other PR, thanks :wink:

[the remark this answers is not in the document]
Comment (msrb) [reply Dockerfile]: @GeetikaBatra ^^

**Pull request merged**